rajawali/src/main/java/org/rajawali3d/view/TextureView.kt

```diff
@@ -0,0 +1,1537 @@
+     *
+     * Use this method to create an OpenGL ES 2.0-compatible context.
+     * Example:
+     * <pre class="prettyprint">
+     * public MyView(Context context) {
+     * super(context);
+     * setEGLContextClientVersion(2); // Pick an OpenGL ES 2.0 context.
+     * setRenderer(new MyRenderer());
+     * }
+    </pre> *
+     *
+     * Note: Activities which require OpenGL ES 2.0 should indicate this by
+     * setting @lt;uses-feature android:glEsVersion="0x00020000" /> in the activity's
+     * AndroidManifest.xml file.
+     *
+     * If this method is called, it must be called before [.setSurfaceRenderer]
+     * is called.
+     *
+     * This method only affects the behavior of the default EGLContexFactory and the
+     * default EGLConfigChooser. If
+     * [.setEGLContextFactory] has been called, then the supplied
+     * EGLContextFactory is responsible for creating an OpenGL ES 2.0-compatible context.
+     * If
+     * [.setEGLConfigChooser] has been called, then the supplied
+     * EGLConfigChooser is responsible for choosing an OpenGL ES 2.0-compatible config.
+     *
+     * @param version The EGLContext client version to choose. Use 2 for OpenGL ES 2.0
+     */
+    fun setEGLContextClientVersion(version: Int) {
+        checkRenderThreadState()
+        eglContextClientVersion = version
+    }
+
+    /**
+     * Inform the view that the activity is paused. The owner of this view must
+     * call this method when the activity is paused. Calling this method will
+     * pause the rendering thread.
+     * Must not be called before a renderer has been set.
+     */
+    fun onPause() {
+        if (rendererDelegate != null) {
+            rendererDelegate!!.mRenderer.onPause()
+        }
+        if (glThread != null) {
+            glThread!!.onPause()
+        }
+    }
+
+    /**
+     * Inform the view that the activity is resumed. The owner of this view must
+     * call this method when the activity is resumed. Calling this method will
+     * recreate the OpenGL display and resume the rendering
+     * thread.
+     * Must not be called before a renderer has been set.
+     */
+    fun onResume() {
+        if (rendererDelegate != null) {
+            rendererDelegate!!.mRenderer.onResume()
+        }
+        glThread!!.onResume()
+    }
+
+    /**
+     * Queue a runnable to be run on the GL rendering thread. This can be used
+     * to communicate with the Renderer on the rendering thread.
+     * Must not be called before a renderer has been set.
+     *
+     * @param r the runnable to be run on the GL rendering thread.
+     */
+    fun queueEvent(r: Runnable) {
+        glThread!!.queueEvent(r)
+    }
+
+    protected class RendererDelegate(internal val mRenderer: ISurfaceRenderer, internal val mRajawaliTextureView: TextureView) : SurfaceTextureListener {
+
+        init {
+            mRenderer.frameRate = if (mRajawaliTextureView.mRenderMode == ISurface.RENDERMODE_WHEN_DIRTY)
```

Comment:
this would look cleaner as

```kotlin
mRenderer.frameRate = when (mRajawaliTextureView.mRenderMode) {
  ISurface.RENDERMODE_WHEN_DIRTY -> mRajawaliTextureView.frameRateTexture
  else -> 0.0
}
```

Reply:
done